// @ilb/filedossier-api/dist/model/Dossier.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * 
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * 
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * 
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * 
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

var _DossierFile = require('./DossierFile');

var _DossierFile2 = _interopRequireDefault(_DossierFile);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
 * The Dossier model module.
 * @module model/Dossier
 * @version 1.0-SNAPSHOT
 */
var Dossier = function () {
  /**
   * Constructs a new <code>Dossier</code>.
   * @alias module:model/Dossier
   * @param code {String} 
   * @param name {String} 
   */
  function Dossier(code, name, valid) {
    _classCallCheck(this, Dossier);

    Dossier.initialize(this, code, name, valid);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Dossier, null, [{
    key: 'initialize',
    value: function initialize(obj, code, name, valid,readonly) {
      obj['code'] = code;
      obj['name'] = name;
      obj['valid'] = valid;
      obj['readonly'] = readonly;
    }

    /**
     * Constructs a <code>Dossier</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Dossier} obj Optional instance to populate.
     * @return {module:model/Dossier} The populated <code>Dossier</code> instance.
     */

  }, {
    key: 'constructFromObject',
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Dossier();

        if (data.hasOwnProperty('code')) {
          obj['code'] = _ApiClient2.default.convertToType(data['code'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient2.default.convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('valid')) {
          obj['valid'] = _ApiClient2.default.convertToType(data['valid'], 'String');
        }
        if (data.hasOwnProperty('readonly')) {
          obj['readonly'] = _ApiClient2.default.convertToType(data['valid'], 'String');
        }
        if (data.hasOwnProperty('dossierFile')) {
          obj['dossierFile'] = _ApiClient2.default.convertToType(data['dossierFile'], [_DossierFile2.default]);
        }
      }
      return obj;
    }
  }]);

  return Dossier;
}();

/**
 * @member {String} code
 */


Dossier.prototype['code'] = undefined;

/**
 * @member {String} name
 */
Dossier.prototype['name'] = undefined;

Dossier.prototype['valid'] = undefined;

Dossier.prototype['readonly'] = undefined; 

/**
 * @member {Array.<module:model/DossierFile>} dossierFile
 */
Dossier.prototype['dossierFile'] = undefined;

exports.default = Dossier;
