// @ilb/filedossier-api/dist/model/DossierFile.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * 
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * 
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * 
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * 
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
 * The DossierFile model module.
 * @module model/DossierFile
 * @version 1.0-SNAPSHOT
 */
var DossierFile = function () {
  /**
   * Constructs a new <code>DossierFile</code>.
   * @alias module:model/DossierFile
   * @param code {String} 
   * @param name {String} 
   */
  function DossierFile(code, name) {
    _classCallCheck(this, DossierFile);

    DossierFile.initialize(this, code, name);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DossierFile, null, [{
    key: 'initialize',
    value: function initialize(obj, code, name) {
      obj['code'] = code;
      obj['name'] = name;
    }

    /**
     * Constructs a <code>DossierFile</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DossierFile} obj Optional instance to populate.
     * @return {module:model/DossierFile} The populated <code>DossierFile</code> instance.
     */

  }, {
    key: 'constructFromObject',
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DossierFile();

        if (data.hasOwnProperty('code')) {
          obj['code'] = _ApiClient2.default.convertToType(data['code'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient2.default.convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('exists')) {
          obj['exists'] = _ApiClient2.default.convertToType(data['exists'], 'Boolean');
        }
        if (data.hasOwnProperty('readonly')) {
          obj['readonly'] = _ApiClient2.default.convertToType(data['readonly'], 'Boolean');
        }
        if (data.hasOwnProperty('required')) {
          obj['required'] = _ApiClient2.default.convertToType(data['required'], 'Boolean');
        }
      }
      return obj;
    }
  }]);

  return DossierFile;
}();

/**
 * @member {String} code
 */


DossierFile.prototype['code'] = undefined;

/**
 * @member {String} name
 */
DossierFile.prototype['name'] = undefined;

/**
 * @member {Boolean} exists
 */
DossierFile.prototype['exists'] = undefined;

/**
 * @member {Boolean} readonly
 */
DossierFile.prototype['readonly'] = undefined;

/**
 * @member {Boolean} required
 */
DossierFile.prototype['required'] = undefined;

exports.default = DossierFile;
