// @ilb/filedossier-api/dist/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DefaultApi = exports.InlineObject = exports.DossierFile = exports.Dossier = exports.ApiClient = undefined;

var _ApiClient = require('./ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

var _Dossier = require('./model/Dossier');

var _Dossier2 = _interopRequireDefault(_Dossier);

var _DossierFile = require('./model/DossierFile');

var _DossierFile2 = _interopRequireDefault(_DossierFile);

var _InlineObject = require('./model/InlineObject');

var _InlineObject2 = _interopRequireDefault(_InlineObject);

var _DefaultApi = require('./api/DefaultApi');

var _DefaultApi2 = _interopRequireDefault(_DefaultApi);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
* Client library of filedossier-api.<br>
* The <code>index</code> module provides access to constructors for all the classes which comprise the public API.
* <p>
* An AMD (recommended!) or CommonJS application will generally do something equivalent to the following:
* <pre>
* var FiledossierApi = require('index'); // See note below*.
* var xxxSvc = new FiledossierApi.XxxApi(); // Allocate the API class we're going to use.
* var yyyModel = new FiledossierApi.Yyy(); // Construct a model instance.
* yyyModel.someProperty = 'someValue';
* ...
* var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
* ...
* </pre>
* <em>*NOTE: For a top-level AMD script, use require(['index'], function(){...})
* and put the application logic within the callback function.</em>
* </p>
* <p>
* A non-AMD browser application (discouraged) might do something like this:
* <pre>
* var xxxSvc = new FiledossierApi.XxxApi(); // Allocate the API class we're going to use.
* var yyy = new FiledossierApi.Yyy(); // Construct a model instance.
* yyyModel.someProperty = 'someValue';
* ...
* var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
* ...
* </pre>
* </p>
* @module index
* @version 1.0-SNAPSHOT
*/
exports.ApiClient = _ApiClient2.default;
exports.Dossier = _Dossier2.default;
exports.DossierFile = _DossierFile2.default;
exports.InlineObject = _InlineObject2.default;
exports.DefaultApi = _DefaultApi2.default; /**
                                            * 
                                            * 
                                            *
                                            * 
                                            * 
                                            *
                                            * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                            * https://openapi-generator.tech
                                            * Do not edit the class manually.
                                            *
                                            */
